### ISSUE TITLE
TypeError: Converting circular structure to JSON in Angluar apps in Preview Mode

### ISSUE THREAD

**Issue by Starker3** [matomo-org/tag-manager#487]
Same type of issue as #185 
However, this is happening on the most recent version of Matomo (4.10.0).
This issue seems to only be happening on sites that use Angular (PHP sites for example do not have the issue using the same container).

![image](https://user-images.githubusercontent.com/20905821/169722889-abf4d06c-079b-4881-897b-c3697140626b.png)

The user was able to reproduce the error by first clicking a button on their site and then clicking on the Event in the preview mode iFrame:
![image](https://user-images.githubusercontent.com/20905821/169722927-b4abef25-90d6-4821-a905-769ba4653f2e.png)

This results in a blank preview page and the **above** error in the console:
![image](https://user-images.githubusercontent.com/20905821/169722941-a510bce7-0cdc-41a4-a3f1-72f0ca7ca71d.png)

I've asked the user to confirm if the same was happening in 4.9 or if it only started after the update to 4.10.0
Will add a comment here if I get confirmation about that.

**Comment by Starker3** [comment]
User downgraded to 4.9 and the same issue is still present.

**Comment by AltamashShaikh** [comment]
@Starker3 I tried replicating the issue by creating a test HTML, but unable to reproduce it
```
<!DOCTYPE html>
<html lang="en-US">
<script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.6.9/angular.min.js"></script>
<body>


<div ng-app="myApp" ng-controller="myCtrl">
    Name: <input ng-model="firstname">
    <h1>{{firstname}}</h1>
</div>

<script>
    var app = angular.module('myApp', []);
    app.controller('myCtrl', function($scope) {
        $scope.firstname = "John";
        $scope.lastname = "Doe";
    });
</script>
<script>
    var _mtm = window._mtm = window._mtm || [];
    _mtm.push({'mtm.startTime': (new Date().getTime()), 'event': 'mtm.Start'});
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.async=true; g.src='{YOUR_CONTAINER_URL}'; s.parentNode.insertBefore(g,s);
</script>
</body>
</html>
```

**Comment by Starker3** [comment]
@AltamashShaikh I'll ask the user if they have any public facing sites that we can try test this on.

**Comment by AltamashShaikh** [comment]
He everyone this is a regression and I was able to identify the cause, will raise a PR for same to fix it

**Comment by mattoii** [comment]
@AltamashShaikh hi friend. Is Bug have been comleted? I dont see changing(

О Matomo 4.10.1, http://joxi.ru/nAyw9pbH9z94ym , http://joxi.ru/J2bG3xDTq6qlBA

**Comment by AltamashShaikh** [comment]
Hey @mattoii It will be fixed with Matomo 4.11
[4.11-rc1](https://github.com/matomo-org/matomo/releases/tag/4.11.0-rc1) is released already

**Comment by mattoii** [comment]
Hey @AltamashShaikh. We installed Matomo 4/11, but preview whatever with error. What i do wrong? 


![image](https://user-images.githubusercontent.com/106577702/173758642-3a8ffccb-ca3a-4dd5-b89b-7f25de3d5d22.png)


![image](https://user-images.githubusercontent.com/106577702/173758050-de87c1af-50b4-45be-a43a-1726e29c2b6e.png)

**Comment by AltamashShaikh** [comment]
@mattoii Can you check if the following file(plugins/TagManager/vue/src/Debugging/Debugging.vue) has identical content on your instance?
File for comparison
`https://github.com/matomo-org/tag-manager/blob/4.x-dev/vue/src/Debugging/Debugging.vue`

**Comment by mattoii** [comment]
@AltamashShaikh yes, what else can i check?

![image](https://user-images.githubusercontent.com/106577702/173833723-a78320dc-88e2-425b-9d4d-1b3a257e602a.png)

**Comment by AltamashShaikh** [comment]
@mattoii is there any way I can test it?

**Comment by AltamashShaikh** [comment]
@mattoii The fix is with `4.11` we completed the fix a bit late due to which its not available in `4.11-rc1` branch.
If you want to test it you might have to update the TagManager plugin to latest code and regenerate the container.

**Comment by mattoii** [comment]
@AltamashShaikh we already installed last version TM
![image](https://user-images.githubusercontent.com/106577702/175013718-6570eca9-325b-462c-8780-1a93bb3d6f35.png)

and we created new tag and installed it. 
![image](https://user-images.githubusercontent.com/106577702/175014034-53a3b455-0d94-4bce-a276-2e99862f022c.png)

and debug still crashed 
![image](https://user-images.githubusercontent.com/106577702/175014310-64e17a04-252a-4199-b144-40c1bf621da9.png)

We need to waiting new version TM or i do wrong?

**Comment by AltamashShaikh** [comment]
@mattoii you have upgraded to `4.11-rc1` which doesn't have this fix, btw how do you update Matomo ?
If via git you can do following things.
1. `cd plugins/TagManager/;`
2. `git checkout 4.x-dev; git pull;`
3. `cd ../../;`
4. `./console tagmanager:regenerate-released-containers`

**Comment by mattoii** [comment]
@AltamashShaikh hi, we updated it through web interface
![image](https://user-images.githubusercontent.com/106577702/175267240-fd6aed06-1eeb-437a-a899-67b704b103ef.png)

Is this enough or should I do something else?

**Comment by AltamashShaikh** [comment]
Hey @mattoii 
Its 4.11-rc1 the fix is in 4.11 and 4.11 should release next week and you can update using the above method
